### PR TITLE
Add warning about setting up pins in custom firmware

### PR DESCRIPTION
**Pull request by WillB97**
Adds documentation to clarify the effects of  https://github.com/srobo/j5/issues/882 for this year's competition.

### CODE REVIEW

File: programming/sr/ruggeduinos/custom_firmware.md [hunk 1]
```diff
@@ -25,6 +25,11 @@ If you're using completely custom firmware, you'll need to tell the kit to ignor
 [Extension of the SR firmware](#extension) {#extension}
 ------------------------------
 
+<div class="warning">
```
Comment (PeterJCLaw) [inline programming/sr/ruggeduinos/custom_firmware.md:28]: To enable markdown parsing inside the HTML block you'll need to annotate it as such:
```suggestion
<div class="warning" markdown="1">
```

File: programming/sr/ruggeduinos/custom_firmware.md [hunk 2]
```diff
@@ -25,6 +25,11 @@ If you're using completely custom firmware, you'll need to tell the kit to ignor
...
+Because the API sets all pins to inputs when the robot is initialised, you cannot set pin modes in the setup function as these will be overridden.
+In order to setup pins you can either create a function on the ruggeduino that you call after your robot is initialised or use the [python functions](/docs/programming/sr/ruggeduinos/) to setup the pins.
```
Comment (PeterJCLaw) [inline programming/sr/ruggeduinos/custom_firmware.md:30]: Maybe worth linking to the relevant section? Just a thought.
```suggestion
In order to setup pins you can either create a function on the ruggeduino that you call after your robot is initialised or use the [python functions](/docs/programming/sr/ruggeduinos/#pin-modes) to setup the pins.
```

**Review by PeterJCLaw — APPROVED** [review]
Approved once the markdown annotation is added.

File: programming/sr/ruggeduinos/custom_firmware.md (hunk 2 above)
Comment (raccube) [inline programming/sr/ruggeduinos/custom_firmware.md:30]: nit

```suggestion
In order to setup pins you can either create a function on the ruggeduino that you call after your robot is initialised or use the [Python functions](/docs/programming/sr/ruggeduinos/) to setup the pins. 
```

**Review by raccube — APPROVED** [review]
Aside from @PeterJCLaw's comments, looks good

File: programming/sr/ruggeduinos/custom_firmware.md (hunk 2 above)
Comment (WillB97) [reply programming/sr/ruggeduinos/custom_firmware.md]: ae69616612f240941c970ec9e52fd55767ab39fb

**Pull request merged**